sw/otbn/crypto/p384_keygen.s

```diff
@@ -0,0 +1,256 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ *   This library contains:
+ *   - P-384 specific routines to generate random values for keys and scalars
+ */
+
+.section .text
+
+/**
+ * Generate a nonzero random value in the scalar field.
+ *
+ * Returns t, a random value that is nonzero mod n, in shares.
+ *
+ * This follows a modified version of the method in FIPS 186-4 sections B.4.1
+ * and B.5.1 for generation of secret scalar values d and k. The computation
+ * in FIPS 186-4 is:
+ *   seed = RBG(seedlen) // seedlen >= 448
+ *   return (seed mod (n-1)) + 1
```

Comment:
nit: there's a new version of this out now since earlier this year -- FIPS 186-5. It specifies basically the same algorithm, with the major difference for P-384 being that the minimum size of seed is now only 384 bits, not 448. Essentially, the curve order for P-384 is so close to 2^384 that the probability of a random 384-bit number being larger than the modulus is less than 2^(-192), and is therefore not a problem for P-384's security.

It's fine with me to merge this as-is, but if you don't mind please create an issue to update the comment to reference FIPS 186-5 and possibly adjust the algorithm to use a smaller seed -- just so we don't forget about it. We'll need to update P-256 as well, since it was written before FIPS 186-5 came out.

Reply:
Alright! I created an issue for this. Thanks!